### PR TITLE
[flutter_tools] wire up native-null-assertions for flutter web

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Add --native-null-assertions to flutter run/drive for web only and to flutter build web, and enable by default for debug and release builds.

https://github.com/flutter/flutter/issues/71614

### CODE REVIEW

File: packages/flutter_tools/lib/src/build_system/targets/web.dart [hunk 1]
```diff
@@ -190,13 +193,16 @@ class Dart2JSTarget extends Target {
   Future<void> build(Environment environment) async {
     final BuildMode buildMode = getBuildModeForName(environment.defines[kBuildMode]);
     final bool sourceMapsEnabled = environment.defines[kSourceMapsEnabled] == 'true';
+    final bool nativeNullAssertions = environment.defines[kNativeNullAssertions] == 'true';
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/build_system/targets/web.dart:196]: Should these have `MissingDefineException` checks?

File: packages/flutter_tools/lib/src/commands/build_web.dart [hunk 2]
```diff
@@ -26,6 +26,7 @@ class BuildWebCommand extends BuildSubCommand {
     usesWebRendererOption();
     addEnableExperimentation(hide: !verboseHelp);
     addNullSafetyModeOptions(hide: !verboseHelp);
+    addNativeNullAssertions(hide: false);
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/commands/build_web.dart:29]: Why is this not hidden but `run` is?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/commands/build_web.dart:29]: For build web, it makes sense 100% of the time. For run, it is only applicable to web targets so I hid it under the verbose options

File: packages/flutter_tools/lib/src/build_system/targets/web.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/build_system/targets/web.dart:196]: I think its okay to assume a default

Comment (jmagman) [reply packages/flutter_tools/lib/src/build_system/targets/web.dart:196]: Works now, could catch something in a refactor in the future.  Up to you though, you are the `assemble` Architect.

**Review by jmagman — APPROVED** [review]
LGTM

**Pull request merged**